src/Clients/IModerationClient.cs

```diff
@@ -0,0 +1,37 @@
+using System.Threading.Tasks;
+using StreamChat.Models;
+
+namespace StreamChat.Clients
+{
+    /// <summary>
+    /// A client that can be used to access moderation endpoints of a Stream Chat application.
+    /// </summary>
+    public interface IModerationClient
+    {
+        /// <summary>
+        /// Check content for moderation.
+        /// </summary>
+        /// <param name="entityType">Type of entity to be checked E.g., stream:user, stream:chat:v1:message, or any custom string</param>
```

Comment:
Docs should mention that `ModerationEntityTypes` contains predefined values of `entityType`. If this accepts predefined values only, it might make more sense to create a custom struct `entityType` so that predefined values are backed by the compiler and code autocompletion. This would simplify things for customers.